test/components/birth-date-picker.spec.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from "react";
+import { ClientRenderer, expect, sinon, waitFor, change } from "test-drive-react";
+import { BirthDatePicker, dateFromYearMonthDay } from "../../src";
+import { BirthDatePickerDemo } from "../../demo/components/birth-date-picker-demo";
+
+describe("<BirthDatePicker />", () => {
+    const clientRenderer = new ClientRenderer();
+    afterEach(() => clientRenderer.cleanup());
+
+    it("Displays the date and allows changing it", async function() {
+        const { select, waitForDom } = clientRenderer.render(<BirthDatePickerDemo />);
+
+        await waitForDom(() => {
+            expect(select("BIRTH_DATE_PICKER")).to.be.present();
+        });
+
+        change(select("BIRTH_DATE_PICKER_YEAR"), "2002");
+        change(select("BIRTH_DATE_PICKER_MONTH"), "10");
+        change(select("BIRTH_DATE_PICKER_DAY"), "12");
+
+        await waitFor(() => {
+            expect(select("BIRTH_DATE_PICKER_DEMO_RESULT")).to.contain.text("Selected date: 2002-10-12");
+        });
+
+        change(select("BIRTH_DATE_PICKER_DAY"), "");
+        await waitFor(() => {
```

Comment:
I think you should use waitForDom instead. Also, I am not sure if it matters when using await, but I think you should return the last call just in case.